pkgs/build-support/vm/default.nix

```diff
@@ -985,25 +1014,287 @@ rec {
 
   # The set of supported RPM-based distributions.
 
-  rpmDistros = { };
+  rpmDistros = {
+    fedora42x86_64 = {
+      name = "fedora-42-x86_64";
+      fullName = "Fedora 42 (x86_64)";
+      packagesList = fetchurl {
+        url = "https://dl.fedoraproject.org/pub/fedora/linux/releases/42/Everything/x86_64/os/repodata/cd483b35df017d68b73a878a392bbf666a43d75db54c386e4720bc369eb5c3a3-primary.xml.zst";
```

Comment:
https://dl.fedoraproject.org/pub/fedora/linux/releases/40/

looks like Fedora deletes files from `dl.fedoraproject.org` once the release goes EOL (so e.g. this URL probably has a lifespan < 1yr at this point) -- and moves them to a separate archive location.

interestingly, Fedora 41 is available both on `dl.fedoraproject.org` _and_ on the archive ([here](https://archives.fedoraproject.org/pub/archive/fedora/linux/updates/41)). i haven't compared those directories, but if the archives are a 1-to-1 copy of whatever we're grabbing, maybe we can paper over the instability by just listing both URLs here?